NumberTheory/GCD.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+    Euclid's GCD algorithm:
+    Basic Euclidean Algorithm is used to find GCD of two numbers say a and b. 
+    The Euclidean algorithm is based on the principle that the greatest common divisor
+    of two numbers does not change if the larger number is replaced by its difference with
+    the smaller number.
+    Below is a recursive C++ function to evaluate gcd using Euclid’s algorithm.
+    
+    Time complexity:
+    O(log10(M)), where M = min(A, B) is the smaller of the input integers A and B
+    
+    Space complexity:
+    Recursive: O(log10(M)), where M = min(A, B) is the smaller of the input integers A and B
```

Comment:
Space complexity is _O(1)_, as the size (space) of internal stack for function calls isn't considered while calculating the space complexity. I missed this point before.